test/jellyfish/component/hls/manager_test.exs

```diff
@@ -38,24 +38,28 @@ defmodule Jellyfish.Component.HLS.ManagerTest do
     hls_dir: hls_dir,
     options: options
   } do
-    create_expect(0)
+    http_mock_expect(0)
+    pid = start_mock_engine()
 
-    {:ok, pid} = Manager.start(room_id, spawn(fn -> nil end), hls_dir, options)
-    ref = Process.monitor(pid)
+    {:ok, manager} = Manager.start(room_id, pid, hls_dir, options)
+    ref = Process.monitor(manager)
 
-    assert_receive {:DOWN, ^ref, :process, ^pid, _reason}
+    kill_mock_engine(pid)
+
+    assert_receive {:DOWN, ^ref, :process, ^manager, :normal}
```

Comment:
Shouldn't spawning manager without credentials be finished with a reason different than `:normal`?

Reply:
Nope. For a reason this is HLS manager not s3 uploader. It manages all the hls processing including removing hls from a disk if needed.